Formula/luau.rb

```diff
@@ -0,0 +1,22 @@
+class Luau < Formula
+  desc "Fast, safe, gradually typed embeddable scripting language derived from Lua"
+  homepage "https://luau-lang.org"
+  url "https://github.com/Roblox/luau/archive/0.504.tar.gz"
+  sha256 "544b3107e1c53f8349b1e327efed55ac29cc20a2983dd23031ab07c5b6944633"
+  license "MIT"
+  revision 0
+  head "https://github.com/Roblox/luau.git", branch: "master"
+  depends_on "cmake" => :build
+
+  def install
+    system "cmake", ".", "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
+    system "cmake", "--build", ".", "--target", "Luau.Repl.CLI", "--config", "RelWithDebInfo"
+    system "cmake", "--build", ".", "--target", "Luau.Analyze.CLI", "--config", "RelWithDebInfo"
+    bin.install "luau", "luau-analyze"
+  end
+
+  test do
+    system "echo \"print ('Homebrew is awesome!')\" >> test.lua"
+    assert_match "Homebrew is awesome!", shell_output("#{bin}/luau test.lua")
```

Comment:
If `luau` reads from stdin you can also use `pipe_output` instead of writing to a file.

Reply:
`luau` does read from stdin, but for some reason, using pipe output doesn't work. The command gives the correct output but does not terminate afterwards. I'm not sure how to fix this so I just used a file instead.